proto/redpanda/api/dataplane/v1/mcp.proto

```diff
@@ -0,0 +1,664 @@
+  map<string, string> tags = 5 [(buf.validate.field).map = {
+    max_pairs: 16
+    values: {
+      string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+    }
+  }];
+}
+
+// CreateMCPServerRequest is the request of CreateMCPServer.
+message CreateMCPServerRequest {
+  MCPServerCreate mcp_server = 1;
+}
+
+message CreateMCPServerResponse {
+  MCPServer mcp_server = 1;
+}
+
+message GetMCPServerRequest {
+  // MCP server ID.
+  string id = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_/]+$"
+  ];
+}
+
+message GetMCPServerResponse {
+  MCPServer mcp_server = 1;
+}
+
+message ListMCPServersRequest {
+  // List filter.
+  Filter filter = 1;
+
+  // Limit the paginated response to a number of items.
+  int32 page_size = 2 [
+    (buf.validate.field).int32 = {
+      gte: -1
+      lte: 1000
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Limit the paginated response to a number of items. Defaults to 100. Use -1 to disable pagination."
+      minimum: -1
+      maximum: 1000
+    }
+  ];
+
+  // Value of the next_page_token field returned by the previous response.
+  // If not provided, the system assumes the first page is requested.
+  string page_token = 3;
+
+  message Filter {
+    // Substring match on MCP server name. Case-sensitive.
+    string name_contains = 1 [
+      (buf.validate.field).ignore = IGNORE_IF_UNPOPULATED,
+      (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+      (buf.validate.field).string.max_len = 128,
+      (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Any MCP Server that partially match this name will be returned."}
+    ];
+
+    // Filter MCP servers that contain all of these key/value pairs.
+    map<string, string> tags = 2 [
+      (buf.validate.field).map = {
+        max_pairs: 16
+        values: {
+          string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+        }
+      },
+      (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "MCP servers that match all the provided tags will be returned."}
+    ];
+  }
+}
+
+message ListMCPServersResponse {
+  repeated MCPServer mcp_servers = 1;
+  string next_page_token = 2;
+}
+
+message MCPServerUpdate {
+  option (google.api.resource) = {
+    type: "redpanda.api.dataplane.v1/MCPServerUpdate"
+    singular: "mcp_server"
+    plural: "mcp_servers"
+  };
+
+  // User-friendly MCP servers name.
+  string display_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+    (buf.validate.field).string.min_len = 3,
+    (buf.validate.field).string.max_len = 128
+  ];
+
+  // MCP server description.
+  string description = 2 [
+    (buf.validate.field).ignore = IGNORE_IF_UNPOPULATED,
```

Comment:
IGNORE_IF_UNPOPULATED is not needed, we ignore validation errors if the update_mask does not include the field